assembl/models/auth.py

```diff
@@ -1989,3 +2128,34 @@ def __repr__(self):
                 LanguagePreferenceOrder(self.source_of_evidence).name,
                 self.preferred_order or 0
             )
+
+
+class PostUserLanguagePreference(UserLanguagePreference):
+    """Does this user wants data in this language to be displayed or translated?"""
+    __tablename__ = 'post_user_language_preference'
+
+    id = Column(Integer, ForeignKey(
+        UserLanguagePreference.id, ondelete='CASCADE', onupdate='CASCADE'),
+        primary_key=True)
+
+    __mapper_args__ = {
+        'polymorphic_identity': 'post_user_language_preference'
+    }
+
+    post_id = Column(Integer, ForeignKey("content.id",
+                     ondelete='CASCADE', onupdate='CASCADE'),
+                     nullable=False, index=False)
+
+    post = relationship("Content",
+                        backref=backref('user_language_preferences'))
+
+    def unique_query(self):
+        query, _ = super(PostUserLanguagePreference, self).unique_query()
+        query = query.filter_by(post_id=self.post_id)
+        return query, True
+
+    # Query a property for fetching a user's preferences on a post
+    def __repr__(self):
+        string_rep = super(PostUserLanguagePreference, self).__repr__()
```

Comment:
You shouldn't call `__repr__()` yourself, use the build-in `repr(super(PostUserLanguagePreference, self))`

Reply:
Fair, are there triggers that get called this way?

Reply:
yes, same for len which call __len__, str which call __str__ etc.